server/s3_common_utilities.cc

```diff
@@ -126,6 +126,52 @@ bool stoi(const std::string &str, int &value) {
   return isvalid;
 }
 
+void size_based_bucketing_of_objects(std::string &oid_str, size_t obj_size){
+  s3_log(S3_LOG_INFO, ""  , "Entering\n");
+  size_t object_size_for_bucketing = obj_size;
+
+  // object size range                           prefix
+
+  // size > 100GB                                 'D'
+  // 1GB < size <= 100GB                          'E'
+  // 50MB < size <= 1GB                           'F'
+  // 10KB < size <= 50MB                          'G'
+  // 1KB < size <= 10KB                           'H'
+  // size <= 1KB                                  'I'
+
+  // NOTE : chars A/B/C have been left out for objects bigger than 100GB
+  // Listing of keys in motr happens in lexicographic order, hence bigger object
+  // will get listed first.
+
+  switch(object_size_for_bucketing){
```

Comment:
Suggestion: My personal experience is switch statements are error prone.

Can we use if-else statements ? 

Like:
if (object_size_for_bucketing > 100GB) {
} else if (object_size_for_bucketing > 1GB && object_size_for_bucketing < 100GB) {
} else if () {

Reply:
I think it is better to use switch case since it is more readable as well as if there are more than five cases in a switch statements, it is implemented using a lookup table/hash list whereas in case of 'if:s' where the last item takes much more time to reach as it has to evaluate every previous condition first.